src/tests/unit/tests/DevTools/target-page-inspector.test.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { TargetPageInspector } from 'Devtools/target-page-inspector';
+import { IMock, It, Mock, Times } from 'typemoq';
+
+describe('TargetPageInspector', () => {
+    type InspectedWindow = typeof chrome.devtools.inspectedWindow;
+
+    let inspectedWindowMock: IMock<InspectedWindow>;
+    let testSubject: TargetPageInspector;
+
+    const testFrameUrl = 'test-frame-url';
+
+    beforeEach(() => {
+        inspectedWindowMock = Mock.ofType<InspectedWindow>();
+        testSubject = new TargetPageInspector(inspectedWindowMock.object);
+    });
+
+    const safeSelectors = [
+        '#id-selector',
+        '.class-selector',
+        'article > p',
+        'article h2',
+        'article ~ h2',
+        'article + h2',
+        'li:last-child',
+        'input:checked + label',
+        'a[target]',
+        'a[href*="login"]',
+        'a[href^="https://"]',
+        'a[href$=".pdf"]',
+        'a[rel~="tag"]',
+        'a[lang|="en"]',
+        'li:nth-child(3n)',
+        'div:not(.awesome)',
+        'div::after',
+        '#result;button',
+    ];
+
+    it.each(safeSelectors)('calls eval on the target page with selector = %s', safeSelector => {
+        const expectedScript =
+            'inspect(document.querySelector(' + JSON.stringify(safeSelector) + '))';
```

Comment:
I feel like it's a little disingenuous for the test to invoke `JSON.stringify`; I think I'd lean more towards pinning the expected results (I think it'd be most readable to use the table template string form of `it.each`, but snapshots would also be ok)

Reply:
Alternately, maybe an approach similar to the below suggestion for the unsafe cases would be a more valuable test case:

* define inspect and document.querySelector in the context of the test (as verifiable mocks)
* have the test literally eval the string passed to inspected.eval
* verify that the document.querySelector mock gets called with the original selector

Reply:
I followed this approach for both safe and unsafe selector. More details on the other comment.